### ISSUE TITLE
Selective Regeneration

### ISSUE THREAD

**Issue by bencoman** [exercism/pharo-smalltalk#191]
Looking at https://github.com/exercism/pharo-smalltalk/pull/189/files#diff-58c1ec3ffc762ccb7009d22063e613e1L91
where it seems only the uuid and "generated from" date has changed,
it would be great if new and old were compared with that metadata filtered out,
and if there are no changes except metadata, leave the file alone.

btw, naive questions... why are the #version numbers different...
e.g. ['2.0.0' here](https://github.com/exercism/pharo-smalltalk/pull/189/files#diff-58c1ec3ffc762ccb7009d22063e613e1L96) and ['2.3.0' here](https://github.com/exercism/pharo-smalltalk/pull/189/files#diff-f1c51904d0328de68627fabc525d80deL58)

**Comment by macta** [comment]
Yes - at the moment I just delete and regenerate over as it’s the easiest thing to do, but now we have an ExercismExercise object that you can easily ask the version number for what’s there and see if it’s changed. That’s part of the crux of the story and it would make it easier to track differences. It’s not hard to do , but I was thinking of cranking a few more exercises and then look at some simple mentoring tools.

**Comment by macta** [comment]
To answer the question about version number changes - if it has changed it’s because exercism have updated the exercise in some way. This round I saw the ordering of some tests changed in some exercises, o d exercise was renamed and another got a few more tests. All those incur a version no. Change and we could have just regenerated those.

**Comment by bencoman** [comment]
No rush for this.  Just recording the thought as it happened.
